bundle/definition/schema.go

```diff
@@ -107,3 +109,34 @@ func (s *Schema) UnmarshalJSON(data []byte) error {
 	}
 	return json.Unmarshal(data, &wrapper)
 }
+
+// ConvertValue attempts to convert the given string value to the type from the
+// definition. Note: this is only applicable to string, number, integer and boolean types
+func (s *Schema) ConvertValue(val string) (interface{}, error) {
+	dataType, ok, err := s.GetType()
+	if !ok {
+		return nil, errors.Wrapf(err, "unable to determine type: %v", s.Type)
+	}
+	switch dataType {
+	case "string":
+		return val, nil
+	case "number":
+		num, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to convert %s to number", val)
+		}
+		return num, nil
+	case "integer":
+		return strconv.Atoi(val)
+	case "boolean":
+		if strings.ToLower(val) == "true" {
```

Comment:
nit: what about ?
```go
switch strings.ToLower(val){
case "true":
    return true, nil
case "false":
    return false, nil
default:
    return false, errors.Errorf("%q is not a valid boolean", val)
}
```

Reply:
Hey @silvin-lubecki I can do a follow-up for this. This was basically just re-introducing the existing code. I like your suggestion though, so I will open a followon.